backend/apps/slack/common/handlers.py

```diff
@@ -0,0 +1,263 @@
+            if leaders and presentation.include_metadata
+            else ""
+        )
+
+        name = Truncator(escape(chapter["idx_name"])).chars(
+            presentation.name_truncation, truncate="..."
+        )
+        summary = Truncator(chapter["idx_summary"]).chars(
+            presentation.summary_truncation, truncate="..."
+        )
+
+        blocks.append(
+            markdown(
+                f"{idx + 1}. <{chapter['idx_url']}|*{name}*>{NL}"
+                f"_{location}_{NL}"
+                f"{leaders_text}"
+                f"{escape(summary)}{NL}"
+            )
+        )
+
+    if presentation.include_feedback:
+        blocks.append(
+            markdown(
+                f"⚠️ *Extended search over {Chapter.active_chapters_count()} OWASP chapters "
+                f"is available at <{get_absolute_url('chapters')}"
+                f"?q={search_query}|{settings.SITE_NAME}>*{NL}"
+                f"{FEEDBACK_CHANNEL_MESSAGE}"
+            )
+        )
+
+    return blocks
+
+
+def projects_blocks(
+    search_query: str = "", limit: int = 10, presentation: EntityPresentation | None = None
+):
+    """Project block for both commands and app home."""
+    from apps.owasp.api.search.project import get_projects
+    from apps.owasp.models.project import Project
+
+    presentation = presentation or EntityPresentation()
+    search_query_escaped = escape(search_query)
+
+    attributes = [
+        "idx_contributors_count",
+        "idx_forks_count",
+        "idx_leaders",
+        "idx_level",
+        "idx_name",
+        "idx_stars_count",
+        "idx_summary",
+        "idx_updated_at",
+        "idx_url",
+    ]
+
+    projects = get_projects(search_query, attributes=attributes, limit=limit)["hits"]
+    if not projects:
+        return [
+            markdown(
+                f"*No projects found for `{search_query_escaped}`*{NL}"
+                if search_query
+                else "*No projects found*{NL}"
+            )
+        ]
+
+    blocks = [
+        markdown(
+            f"{NL}*OWASP projects that I found for* `{search_query_escaped}`:{NL}"
+            if search_query_escaped
+            else f"{NL}*OWASP projects:*{NL}"
+        ),
+    ]
+
+    for idx, project in enumerate(projects):
+        name = Truncator(escape(project["idx_name"])).chars(
+            presentation.name_truncation, truncate="..."
+        )
+        summary = Truncator(project["idx_summary"]).chars(
+            presentation.summary_truncation, truncate="..."
+        )
+
+        metadata = []
+        if presentation.include_metadata:
+            if project["idx_contributors_count"]:
+                metadata.append(f"Contributors: {project['idx_contributors_count']} ")
+            if project["idx_forks_count"]:
+                metadata.append(f"Forks: {project['idx_forks_count']}")
+            if project["idx_stars_count"]:
+                metadata.append(f"Stars: {project['idx_stars_count']} ")
```

Comment:
Let's keep it as is but I'm not really sure if we want different presentation.
I appreciate the deep understanding of current differences.